frontend/src/metabase/containers/AddToDashSelectDashModal.unit.spec.tsx

```diff
@@ -0,0 +1,235 @@
+import React from "react";
+import fetchMock from "fetch-mock";
+import { Route } from "react-router";
+import {
+  setupMostRecentlyViewedDashboard,
+  setupCollectionsEndpoints,
+} from "__support__/server-mocks";
+
+import {
+  fireEvent,
+  renderWithProviders,
+  screen,
+  waitForElementToBeRemoved,
+} from "__support__/ui";
+import {
+  createMockCard,
+  createMockCollection,
+  createMockDashboard,
+  createMockUser,
+} from "metabase-types/api/mocks";
+import { Collection, Dashboard } from "metabase-types/api";
+import { AddToDashSelectDashModal } from "./AddToDashSelectDashModal";
+
+const card = createMockCard({ id: 1, name: "Model Uno", dataset: true });
+
+const CURRENT_USER = createMockUser({
+  id: 1,
+  personal_collection_id: 100,
+  is_superuser: true,
+});
+
+const DASHBOARD = createMockDashboard({
+  id: 3,
+  name: "Test dashboard",
+  collection_id: 2,
+  model: "dashboard",
+});
+
+const dashboards = {
+  [DASHBOARD.id]: DASHBOARD,
+};
+
+const COLLECTION_1 = createMockCollection({
+  id: 1,
+  name: "C1",
+  can_write: true,
+});
+
+const COLLECTION_2 = createMockCollection({
+  id: 2,
+  name: "C2",
+  can_write: true,
+});
+
+const PERSONAL_COLLECTION = createMockCollection({
+  id: CURRENT_USER.personal_collection_id,
+  name: "My personal collection",
+  personal_owner_id: CURRENT_USER.id,
+});
+
+const COLLECTIONS = [COLLECTION_1, COLLECTION_2, PERSONAL_COLLECTION];
+
+interface SetupOpts {
+  collections?: Collection[];
+  error?: string;
+  dashboard?: Dashboard;
+  noRecentDashboard?: boolean;
+  waitForContent?: boolean;
+}
+
+const setup = async ({
+  collections = COLLECTIONS,
+  dashboard = DASHBOARD,
+  noRecentDashboard = false,
+  error,
+  waitForContent = true,
+}: SetupOpts = {}) => {
+  setupCollectionsEndpoints(collections);
+  mockCollectionItemsEndpoint([dashboard]);
+  mockRootCollectionItemsEndpoint(collections, [dashboard]);
+  mockCollectionByIdEndpoint({ collections: COLLECTIONS, error });
+  setupMostRecentlyViewedDashboard(noRecentDashboard ? undefined : dashboard);
+
+  renderWithProviders(
+    <Route
+      path="/"
+      component={() => (
+        <AddToDashSelectDashModal
+          card={card}
+          onChangeLocation={() => undefined}
+          onClose={() => undefined}
+          dashboards={dashboards}
+        />
+      )}
+    />,
+    {
+      withRouter: true,
+      storeInitialState: {
+        currentUser: CURRENT_USER,
+      },
+    },
+  );
+
+  if (waitForContent) {
+    await waitForElementToBeRemoved(() => screen.queryByText("Loading..."));
+  }
+};
+
+describe("AddToDashSelectDashModal", () => {
+  describe("Create new Dashboard", () => {
+    it("should open CreateDashboardModal", async () => {
+      await setup();
+
+      const createNewDashboard = screen.getByRole("heading", {
+        name: /create a new dashboard/i,
+      });
+
+      fireEvent.click(createNewDashboard);
```

Comment:
Please user `user-event` everywhere instead of `fireEvent`

Reply:
it turned out userEvent's click is just a wrapper for fireEvent, which led to the the worse performance in some cases https://github.com/jsdom/jsdom/issues/3234#issuecomment-1359965734

but I agree userEvent seems to cover more cases, I'll change it